eox_core/api/v1/serializers.py

```diff
@@ -79,6 +85,84 @@ def validate(self, attrs):
+    microsite settings.
+    """
+    year_of_birth = serializers.ChoiceField(choices=YEAR_OF_BIRTH_CHOICES)
+    gender = serializers.ChoiceField(choices=get_gender_choices())
+    city = serializers.CharField()
+    goals = serializers.CharField()
+    bio = serializers.CharField(max_length=3000)
+    profile_image_uploaded_at = serializers.DateTimeField()
+    phone_number = serializers.CharField(max_length=50)
+    mailing_address = serializers.CharField()
+    courseware = serializers.CharField(max_length=255)
+    level_of_education = serializers.ChoiceField(choices=get_level_of_education_choices())
+    country = CountryField(required=False)
+
+    def __init__(self, *args, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Add the custom registration fields specified in the settings.
+        """
+        super(EdxappExtendedUserSerializer, self).__init__(**kwargs)
+
+        extended_profile_fields = getattr(settings, 'extended_profile_fields', [])
+        extra_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+        ednx_custom_regitration_fields = getattr(settings, 'EDNX_CUSTOM_REGISTRATION_FIELDS', [])
+        copy_fields = self.fields
+        extra_fields.pop("terms_of_service", None)
+        copy_fields.pop("activate_user", None)
+        profile_fields = copy_fields.keys() - EdxappUserSerializer().fields.keys()
+
+        # Delete the profile fields that are not allowed or are redefined in the ednx_custom_registration
+        # In case the field IS allowed, check if is required
+        for field in profile_fields:
+            if field not in extra_fields or field in extended_profile_fields:
+                self.fields.pop(field)
+            else:
+                self.fields[field].required = extra_fields.get(field) == "required"
+
+        # Adding fields that go inside the UserProfile.meta
+        for custom_field in ednx_custom_regitration_fields:
+            field_name = custom_field.get("name")
+            field_type = custom_field.get("type")
+
+            if field_name in extended_profile_fields and field_type in ALLOWED_TYPES:
+                serializer_field = {}
+
+                serializer_field["required"] = extra_fields.get(field_name) == "required"
+
+                if field_type in ["text"]:
+                    for key, value in custom_field.get("restrictions", {}).items():
+                        try:
+                            ["min_length", "max_length"].index(key)
+                            serializer_field[key] = int(value)
+                        except ValueError:
+                            raise serializers.ValidationError({"restriction error": "{key}: {value}.\
+                            The restriction may not be valid or the value is not an integer".format(
+                                key=key,
+                                value=value,
+                            )})
+
+                if field_type in ["select"]:
```

Comment:
why not elif? so this is not checked when field_type is text. Or do you need to check it?

Reply:
@mariajgrimaldi  I need to check it because in case the field is of type 'text' It needs to go to the _else_ part (line 162), so it can add the _serializer.Charfield()_

In line 139 I'm only checking if there are any restrictions that are only allowed for text fields, according to this  https://github.com/eduNEXT/edunext-platform/blob/72229d4794a7bcf7a350d686f3d506aa24c1ec15/openedx/core/djangoapps/user_api/helpers.py#L100